### ISSUE TITLE
Encoding of fermata

### ISSUE THREAD

**Issue by lpugin** [rism-digital/pae-code-spec#3]
The current specifications of PAE support the encoding of fermata with the `(` and `)` character.

From the current specifications

> fermata (include only one note or rest; accidentals or octave symbols must be outside the parentheses. See also "Special rhythmic groupings" below.)

The main problems with this a approach are:
* fermata uses the same characters as the special rhythmic groupings (tuplets), which causes some additional validation / parsing problems. Only the analysis of the content gives the meaning of the signs, which is not very good practice.
* there are some contradictions with trill and tie encoding when saying that only a note or a rest should be included since the trill and tie characters are supposed to follow the note immediately. In other words, encoding a note with a fermata and a trill is currently impossible when following strictly the current PAE specifications.

The proposal would be to use a single characters for encoding fermata since there is no need to have a start and end delimiter. It would avoid overlap with tuplets, it would also work better with chords and would be more inline with the current encoding of trills.

Some possible characters could be:
* `?`
* `*`
* `c` (corona)
* `h` (hold)
* `p` (pause and point d'orgue)

Placement of the sign should be after the note / chord

**Comment by ahankinson** [comment]
Trilled chords seem out of scope to me…

**Comment by lpugin** [comment]
Maybe. We have about only 10 of them https://rism.online/sources/300000052/. As long as it is clearly specified, I don't see a problem with them, though.

**Comment by ahankinson** [comment]
#110 would get weird if we had tied and trilled chords…

**Comment by ahankinson** [comment]
Also, the fermata modifies the duration of the note, so it makes sense that it goes before the note name?

**Comment by lpugin** [comment]
> Also, the fermata modifies the duration of the note, so it makes sense that it goes before the note name?

Yes, I agree. But then would `4pCEG` be 

1 <img width="163" alt="image" src="https://github.com/rism-digital/pae-code-spec/assets/689412/7fa3a79e-f5d7-4ac2-8474-7253289526e3"> or 2 <img width="163" alt="image" src="https://github.com/rism-digital/pae-code-spec/assets/689412/a8944795-b796-465a-a9e6-37a119cd0f3f"> ?

If you see it bound to the duration, it should be 1., otherwise it should be placed after the note IMO. And I see 2. to be less of a change with V1, where I don't think a major reason to go for 1. - both have valid arguments.

**Comment by ahankinson** [comment]
> If you see it bound to the duration, it should be 2., otherwise it should be placed after the note

Sure, but we haven't really established such a concept -- it's not really "bound" to the duration. The `g` and `q` also comes before the note, but it only lasts for one note. There's no expectation that it translates to the following notes.

We can also move those after the notes if we want to introduce the idea that "inherited" attributes come before the note name, while "note-specific" attributes come after?

**Comment by ahankinson** [comment]
> PS V1 does 4x(F), changing to 4xFp makes more sense than changing to 4pxF

As they are written now, the rules say the fermata must be immediately before the note name, and after all other attributes. So it would be `4xpF`, which is just as easy to convert as `4xFp`.

**Comment by lpugin** [comment]
Yes, so it does not look like it is modifying the duration. It is modifying the note, and as I proposed initially it should be placed after. I am not sure where the problem is with this. Could you explain why you decided against it?

**Comment by ahankinson** [comment]
I didn't decide against it... I just had to choose one or the other. I'm ok moving it, but I think we should establish a pattern for this so that it's not just arbitrary about which things come before, and which come after.

**Comment by lpugin** [comment]
OK, let's move it after. The pattern I have in mind is:
* duration and pitch codes before the note
* modifiers (trill, fermata, ties) codes after

**Comment by ahankinson** [comment]
OK, sounds good to me.

**Comment by ahankinson** [comment]
Modelling this out, I think we'll run into a problem with appogiatura. Consider:

`8Aq{8CD}E`

![image](https://github.com/rism-digital/pae-code-spec/assets/163183/7c0aa092-11b4-43e7-80d0-66842cd73358)

and 

`8Aqq{8CD}rE`

![image](https://github.com/rism-digital/pae-code-spec/assets/163183/4a5318d7-eb5d-4f65-b36c-3de2e2674732)

The first adds the `q` at the end of the note (based on this conversation), but the second starts a new grace group for the notes that are following. If you're parsing the string, it's difficult to know, until the second `q`, that the appogiatura applies to the current note, or if it's the start of an appogiatura group. 

Potential solutions:

 - Change the appogiatura character
 - Change the appogiatura group start character(s)
 - Live with the uncertainty
 - Keep the appogiatura at the beginning

Personally, I might go for the second option, and change appogiatura groups to start with a letter like `t` or `s`, e.g.,

``8At{8CD}rE`

**Comment by lpugin** [comment]
I would keep the appogiatura at the beginning. We don't want to change too much when it is not strictly needed.

**Comment by ahankinson** [comment]
`qq` is a weird one, though. It's the only grouping mechanism that uses two characters instead of one. 

Interestingly, the original papers had `q` as the start of the group, and `qq` as the cancellation. 

![image](https://github.com/rism-digital/pae-code-spec/assets/163183/9aac0b63-ece6-4375-aa03-5829f05409f6)

**Comment by ahankinson** [comment]
Also important is that it didn't distinguish between acciaccatura and appogiatura.

**Comment by lpugin** [comment]
I completely agree. As a matter of fact, one thing the Verovio parser does is replace qq with Q before doing any parsing.

Maybe we can change qq to s, but I would definitely keep q before the note.

**Comment by lpugin** [comment]
Or y so every grace note start has a glyph going under the baseline?

**Comment by ahankinson** [comment]
I like `y` as the start of a group.

Not sure where that leaves us with any sort of logical order with note attributes, though. It would be nice to have some way of explaining their order, instead of it being arbitrary.

**Comment by lpugin** [comment]
I see this as a major, more duration-like, qualifier,  which is why it comes before. This is different from t or p.

**Comment by lpugin** [comment]
Sure. Let's talk about it. But we shouldn't try to draw too hard lines when trying to organise things as long as we can show some coherence. Being too strict in domains and concepts never works with music notation.

**Comment by ahankinson** [comment]
Yeah, it's the "some" coherence I'm worried about. Right now it all seems a bit arbitrary.

**Comment by lpugin** [comment]
Sure. Let's talk about it. But we shouldn't try to draw too hard lines when trying to organise things as long as we can show some coherence. Being too strict in domains and concepts never works with music notation. 

Maybe another way to look at this is to categories them as primary modifiers (duration, octave, accidental, grace) essential for basic rendering, and secondary ones (trill, fermata)

**Comment by ahankinson** [comment]
Symbols above the note? That seems to be what fermata and trills are. Maybe, if you squint hard enough, ties would fit in that class too?

**Comment by BaMikusi** [comment]
I actually think that @lpugin's argument about the gracenote being "a major, more duration-like, qualifier" makes sense. In the case of `g` it certainly is, for this character tells you that the following note will not need a strictly measured duration, and should instead be rendered as a really short one -- so here the `g` sort of functions as a substitute for an explicit duration value. But if the `g` comes before the pitch, the `q` should arguably do the same -- and one can in fact also positively argue that here, too, the `q` qualifies the following duration from the start, making clear that it should not strictly be taken at face value.

I of course understand @ahankinson's point that a fermata also affects the duration in the end, but the effect of that seems quite different in that in this case (I am of course speaking of the modern text-book interpretation of fermate here) the explicitly stated duration is absolutely relevant, the note is held that long -- and _after_ that moment comes a (more or less) unexpected turn, namely that the note does not end there, but is held even longer. Viewing the music in this way as a process, the fermata sign seems to become relevant only _after_ the pitch -- quite similarly to a tie, which in certain constellations indeed has a rather similar psychological effect.

I won't argue that this is a waterproof train of thought, but I also have the clear impression that the fermata should come only after, and I think this is the actual musical reason for my strongly feeling so.
